subsys/bluetooth/mesh/light_xyl_srv.c

```diff
@@ -617,6 +609,23 @@ const struct bt_mesh_model_cb _bt_mesh_light_xyl_srv_cb = {
 	.reset = bt_mesh_light_xyl_srv_reset,
 };
 
+static int bt_mesh_light_xyl_setup_srv_init(struct bt_mesh_model *model)
+{
+	struct bt_mesh_light_xyl_srv *srv = model->user_data;
+	int err;
+
+	err = bt_mesh_model_extend(model, srv->model);
+	if (err) {
+		return err;
+	}
+
+	return bt_mesh_model_extend(model, srv->lightness_srv->setup_model);
```

Comment:
This'll fail, I believe. The lightness server isn't instantiated by the xyL macro any more (since #5213), and may sit on a later element. If it does, `lightness_srv->setup_model` hasn't been set yet. 

This'll either need to be moved to the setup function, or be replaced by model_find instead. 

I believe this applies to xyL, light control and HSL, which all take the lightness_server as a pointer. Light control already implements this, but xyL and HSL need to change

Reply:
`light_hsl_srv.rst` recommends putting Lightness Server on the same element as Light HSL. Do you mean that Lightness Server may sit on the same element but later than Light HSL/XYL Server? Or that it can be on another element? I think it can be on any element so model find won't work anyway.